sympy/matrices/determinant.py

```diff
@@ -479,6 +481,59 @@ def _cofactor_matrix(M, method="berkowitz"):
     return M._new(M.rows, M.cols,
             lambda i, j: M.cofactor(i, j, method))
 
+def _per(M):
+    """Returns the permanent of a matrix. Unlike determinant,
+    permanent is defined for both square and non-square matrices.
+
+    For an m x n matrix, with m less than or equal to n,
+    it is given as the sum over the permutations s of size
+    less than or equal to m on [1, 2, . . . n] of the product
+    from i = 1 to m of M[i, s[i]]. Taking the transpose will
+    not affect the value of the permanent.
+
+    In the case of a square matrix, this is the same as the permutation
+    definition of the determinant, but it does not take the sign of the
+    permutation into account. Computing the permanent with this definition
+    is quite inefficient, so here the Ryser formula is used.
+
+    Examples
+    ========
+    >>> from sympy import Matrix
+    >>> M = Matrix([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+    >>> M.per()
+    450
+    >>> M = Matrix([1, 5, 7])
+    >>> M.per()
+    13
+    References
+    ==========
```

Comment:
```suggestion
    ==========

```
I'd also prefer to add spacings below every section titles

Reply:
Yes, I've added them. On a sidenote, is there any way to run the build locally and push changes once it is done? I'm asking because that way no. of failed builds would reduce

Reply:
I'd refer to https://github.com/sympy/sympy/wiki/Running-tests for how to run tests.
You can also use pytest

Reply:
The build ran correctly, but errored while building the docs. Anything to do with the code I wrote?